IPython/core/tests/test_interactiveshell.py

```diff
@@ -561,6 +561,20 @@ def test_exit_code_signal_csh(self):
             else:
                 del os.environ['SHELL']
 
+    def test_removes_temp_files_and_directories(self):
+        # First we ensure that we have created at least a tempfile
+        ip.mktempfile()
+
+        self.system("exit 0")
+        ip.atexit_operations()
+
```

Comment:
@Carreau I'm not sure how to test this case...
I want to be sure that every time a shell is closed we are removing the temp dirs/files.
Any thoughts?

Reply:
Apologies, I received many PR during hacktoberfest, I'll have a look.

Reply:
No problem :+1: